src/evaluation_system/api/parameters.py

```diff
@@ -809,10 +811,105 @@ class MyPlugin(plugin.PluginAbstract):
             )
 
     """
+    base_type = datetime
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+    def parse(self, value: str) -> Union[datetime, None]:
+        """
+        Parse a string value into a datetime object.
+
+        Parameters
+        ----------
+        value : str
+            The string value representing a date.
+
+        Returns
+        -------
+        Union[datetime, None]
+            The parsed datetime object if the parsing is successful,
+            otherwise returns None.
+
+        Raises
+        ------
+        ValueError
+            If the value cannot be converted to a datetime object.
+
+        """
+
+        try:
+            if value:
```

Comment:
since Date is being called from DateRange but also works stand alone I was assuming two behaviours;

a) comming from DateRange is allows to parse the empty ini_date or end_date (for example you want to evaluate the data up to that year, or from a year to the end).

b) if used stand alone supose that you use the Date as a parameter where you define a particular datetime to analyse (for example if you ahve and IDF curve superimpose the return period of a particular event ). But you could also let it empty if you do not want to do that. 

But maybe I did that because I do not know whether if mandatory=false it would still fail if there is no value given to the parameter.

Reply:
I see, so in a nutshell this is because the DateRange class doesn't inherit from Date. Which I think it should. Then you won't have to used this construct. Neither this ` self.dateparse = Date() `

Reply:
DateRange() is maybe not the best choice of name, I just wnated to parse a `"ini_date,end_date"` into a `(datetime(ini_date),datetime(end_date))`
I could call it DatePeriod then